src/main/java/seedu/address/logic/commands/SortEventCommand.java

```
@@ -0,0 +1,128 @@
+    //            }
+    //            if (o1.getTags().size() != 0 && o2.getTags().size() == 0) {
+    //                return -1;
+    //            }
+    //            if (o1.getTags().size() == 0 && o2.getTags().size() == 0) {
+    //                return o1.getName().fullName.compareToIgnoreCase(o2.getName().fullName);
+    //            }
+    //            return o1.getTags().iterator().next().tagName
+    //                    .compareToIgnoreCase(o2.getTags().iterator().next().tagName);
+    //        }
+    //    };
+
+    private final Index index;
+
+    /**
+     * @param index the order in which to sort the address book
+     */
+    public SortEventCommand(Index index) {
+        requireAllNonNull(index);
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        model.sortEvent(chooseComparator(index));
+
+        return new CommandResult(indexMessage(index));
+    }
+
+    /**
+     * Returns the appropriate console message for the index.
+     */
+    public String indexMessage(Index index) {
+        int input = index.getOneBased();
+        switch (input) {
+        case 1:
+            return "Sorted by description in alphabetical order";
+        case 2:
+            return "Sorted by time in chronological order";
+        case 3:
+            return "Sorted by the number of participants in descending order";
+        default:
+            return "Invalid index entered, refer to below for the command's proper usage: "
+                    + MESSAGE_USAGE;
+        }
+    }
+
+    /**
+     * Returns the appropriate comparator for the model manager to sort with
+     */
+    public Comparator<Event> chooseComparator(Index index) {
+        int input = index.getOneBased();
+        switch (input) {
+        case 2:
+            return TIME_COMPARATOR;
+        default:
+            return DESCRIPTION_COMPARATOR;
```

this currently leads to bugs; if i put an index of 3, it'll still return the `DESCRIPTION_COMPARATOR`. i think you should have individual cases for each comparator, and then throw a `ParseException` in the default line.

---

although i think it would be better to have the detection of invalid inputs handled in `SortEventCommandParser.parse`.

---

> although i think it would be better to have the detection of invalid inputs handled in SortEventCommandParser.parse.

the `SortEventCommandParser.parse` is already implemented. But I do agree with the suggestion regarding the default branch. I think it is in-line with defensive programming to reserve the default block to throw an exception instead in order to protect others who might be using the code.

---

yeah so i think the `default` branch in here should throw something like an `IllegalArgumentException` (for us to see, since this branch should never be reached), and the one in `SortEventCommandParser.parse` should throw a `ParseException` (for the user to see, if they input an invalid number).

---

Have removed the third implementation for now as I've already implemented the feature with the tag in another branch.